uitools/widgets/Esri/ArcGISRuntime/Toolkit/FloorFilter.cpp

```diff
@@ -0,0 +1,313 @@
+    QFrame(parent),
+    m_controller(new FloorFilterController(this)),
+    m_ui(new Ui::FloorFilter)
+  {
+    m_ui->setupUi(this);
+
+    m_ui->allSites->setChecked(!m_controller->isSelectedSiteRespected());
+    m_ui->sitesView->setModel(wrapFilterModel(m_controller->sites(), m_ui->sitesFiltler, this));
+    m_ui->facilitiesView->setModel(wrapFilterModel(m_controller->facilities(), m_ui->facilitiesFilter, this));
+    m_ui->levelsView->setModel(m_controller->levels());
+
+    // Changes the contents of the facilities list.
+    connect(
+        m_ui->allSites, &QCheckBox::clicked, this, [this](bool checked)
+        {
+          m_controller->setIsSelectedSiteRespected(!checked);
+          if (checked)
+            m_ui->toolBox->setCurrentIndex(1);
+        });
+
+    // Sites setup
+    connect(
+        m_controller, &FloorFilterController::selectedSiteIdChanged, this, [this](QString /*oldId*/, QString newId)
+        {
+          // Set `m_sitesUpdatedFromController` for the duration of scope, then set back to false.
+          auto b = push_value(m_sitesUpdatedFromController, true);
+          const auto i = indexForId<FloorFilterSiteItem>(m_ui->sitesView->model(), newId);
+          m_ui->sitesView->selectionModel()->setCurrentIndex(i, QItemSelectionModel::SelectCurrent);
+        });
+
+    connect(
+        m_ui->sitesView, &QListView::doubleClicked, this, [this](QModelIndex index)
+        {
+          if (index.isValid())
+            m_ui->toolBox->setCurrentIndex(1);
+        });
+
+    connect(
+        m_ui->sitesView->selectionModel(), &QItemSelectionModel::currentChanged, this, [this](QModelIndex index, QModelIndex previous)
+        {
+          if (index == previous || m_facilitiesUpdatedFromController)
+            return;
+
+          if (index == QModelIndex{})
+          {
+            m_controller->setSelectedSiteId("");
+          }
+          else
+          {
+            const auto data = itemForIndex<FloorFilterSiteItem>(m_ui->sitesView->model(), index);
+            m_controller->setSelectedSiteId(data->modelId());
+            m_controller->zoomToSite(data);
+          }
+        });
+
+    // Facilities setup.
+    connect(
+        m_controller, &FloorFilterController::selectedFacilityIdChanged, this, [this](QString /*oldId*/, QString newId)
+        {
+          auto b = push_value(m_facilitiesUpdatedFromController, true);
+          const auto i = indexForId<FloorFilterFacilityItem>(m_ui->facilitiesView->model(), newId);
+          m_ui->facilitiesView->selectionModel()->setCurrentIndex(i, QItemSelectionModel::SelectCurrent);
+        });
+
+    connect(
+        m_ui->facilitiesView, &QListView::doubleClicked, this, [this](QModelIndex index)
+        {
+          if (index.isValid())
+            m_ui->toolBox->setCurrentIndex(2);
+        });
+
+    connect(
+        m_ui->facilitiesView->selectionModel(), &QItemSelectionModel::currentChanged, this, [this](QModelIndex index, QModelIndex previous)
```

Comment:
this is a massive function with all of the connection lambdas. do you think breaking these out to named slots would aid readability?